autoscheduler/frontend/src/components/SchedulingPage/CourseSelectColumn/CourseSelectCard/ExpandedCourseCard/ExpandedCourseCard.tsx

```diff
@@ -14,6 +14,39 @@ import { CustomizationLevel, CourseCardOptions } from '../../../../../types/Cour
 import { RootState } from '../../../../../redux/reducer';
 import { updateCourseCard, removeCourseCard } from '../../../../../redux/actions/courseCards';
 
+
+function is_number(input_char) {
+//returns true if input is 0-9
+  if(input_char >= "0" && input_char <= "9") {
+    return true;
+  }
+  return false;
+}
+
+function remove_all_spaces(input_string) {
```

Comment:
Originally, this function was remove duplicate spaces replaced with " " instead of "". However, despite working in the W3 try it editor and a few other online javascript compilers, it didn't work on the test build of the website so I went with remove all spaces because for some reason that did and I couldn't find out why. It was not ideal. This may have unintended side effects on title_search so yall may want to silence or remove it. This function only makes it so that autocomplete can work with id_search independent of the number of spaces the user types. However, the add_space_between_course_subject_and_number function below will still work if the user types in 0 or 1 spaces between the subject and course number. Any guidance on why remove_duplicates_spaces did not work would be greatly appreciated.